pygsti/extras/errorgenpropagation/propagatableerrorgen.py

```diff
@@ -0,0 +1,363 @@
+        if self.errorgen_type in 'HS':
+            return self.error_rate*change_basis(create_elementary_errorgen(self.errorgen_type,paulis[0]),'std',matrix_basis)
+        else:
+            return self.error_rate*change_basis(create_elementary_errorgen(self.errorgen_type,paulis[0],paulis[1]),'std',matrix_basis)     
+
+
+
+
+
+'''
+Returns the Commutator of two errors
+'''
+def commute_errors(ErG1,ErG2, weightFlip=1.0, BCHweight=1.0):
+    def com(p1,p2):
+        P1 = pyGSTiPauli_2_stimPauli(p1)
+        P2=pyGSTiPauli_2_stimPauli(p2)
+        P3=P1*P2-P2*P1
+        return (P3.weight,stimPauli_2_pyGSTiPauli(P3))
+    
+    def acom(p1,p2):
+        P1 = pyGSTiPauli_2_stimPauli(p1)
+        P2=pyGSTiPauli_2_stimPauli(p2)
+        P3=P1*P2+P2*P1
+        return (P3.weight,stimPauli_2_pyGSTiPauli(P3))
+    
+    def labelMultiply(p1,p2):
+        P1 = pyGSTiPauli_2_stimPauli(p1)
+        P2=pyGSTiPauli_2_stimPauli(p2)
+        P3=P1*P2
+        return (P3.weight,stimPauli_2_pyGSTiPauli(P3))
+    
+    errorGens=[]
+    
+    wT=ErG1.getWeight()*ErG2.getWeight()*weightFlip*BCHweight
+    
+    if ErG1.getType()=='H' and ErG2.getType()=='H':
+        pVec=com(ErG1.getP1() , ErG2.getP2())
+        errorGens.append( propagatableerrorgen( 'H' , [pVec[1]] , -1j*wT *pVec[0] ) )
+        
+    elif ErG1.getType()=='H' and ErG2.getType()=='S':
+        pVec=com(ErG2.getP1() , ErG1.getP1())
+        errorGens.append( propagatableerrorgen( 'C' , [ErG2.getP1() , pVec[1]] , 1j*wT*pVec[0] ) )
+         
+    elif ErG1.getType()=='S' and ErG2.getType()=='H':
+        pVec=com(ErG2.getP1() , ErG1.getP1())
+        errorGens.append( propagatableerrorgen( 'C' , [ErG2.getP1() , pVec[1]] , -1j*wT *pVec[0] ) )
+          
+    elif ErG1.getType()=='H' and ErG2.getType()=='C':
+        pVec1=com(ErG2.getP1() , ErG1.getP1())
+        errorGens.append( propagatableerrorgen('C' , [pVec1[1], ErG2.getP2()] , 1j*wT*pVec1[0] ) )
+        pVec2=com(ErG2.getP2() , ErG1.getP1())
+        errorGens.append( propagatableerrorgen('C' , [pVec2[1] , ErG2.getP1()] , 1j*wT*pVec2[0] ) )
+                          
+    elif ErG1.getType()=='C' and ErG2.getType()=='H':
+        errorGens = commute_errors(ErG2,ErG1,weightFlip=-1.0,BCHweight=BCHweight)
+    
+    elif ErG1.getType()=='H' and ErG2.getType()=='A':
+        pVec1 = com(ErG1.getP1() , ErG2.getP1())
+        errorGens.append( propagatableerrorgen('A'  , [pVec1[1] , ErG2.getP2()] , -1j*wT*pVec1[0]) )
+        pVec2 = com(ErG1.getP1() , ErG2.getP2())
+        errorGens.append( propagatableerrorgen('A'  , [ErG2.getP1(), pVec2[1]] , -1j*wT*pVec2[0] ) )
+                          
+    elif ErG1.getType()=='A' and ErG2.getType()=='H':
+        errorGens = commute_errors(ErG2,ErG1,weightFlip=-1.0,BCHweight=BCHweight)
+
+    elif ErG1.getType()=='S' and ErG2.getType()=='S':
+        errorGens.append( propagatableerrorgen('H', ErG1.getP1(),0 ))
+                         
+    elif ErG1.getType()=='S' and ErG2.getType()=='C':
```

Comment:
Notes have

$$\large
[S_P, C_{A,B}] = -iA_{AP,BP} -iA_{PB,AP} -\frac{i}{2}\big(A_{\{A,B\}P,P} + A_{P,P\{A,B\}}\big)
$$

but code is 

$$\large
[S_P, C_{A,B}] = -iA_{\textcolor{red}{PA},BP} -iA_{\textcolor{red}{PB},\textcolor{red}{PA}} -\frac{i}{2}\big(A_{\{A,B\}P,P} + A_{P,P\{A,B\}}\big)
$$

Similar to the commutators, this change in multiplication could come with a sign change (depending on whether $[P,A]  = 0$ and $[P,B] = 0$).
(Also we are computing $\{A,B\}$ twice).